core/meterer/offchain_store.go

```diff
@@ -237,3 +237,79 @@ func (s *OffchainStore) GetRelevantOnDemandRecords(ctx context.Context, accountI
 
 	return prevPayment, nextPayment, nextDataLength, nil
 }
+
+func (s *OffchainStore) GetBinUsages(ctx context.Context, accountID string, binIndex uint32) (uint64, uint64, uint64, error) {
+	// Fetch the 3 bins start from the current bin
+	queryInput := &dynamodb.QueryInput{
+		TableName:              aws.String(s.reservationTableName),
+		KeyConditionExpression: aws.String("AccountID = :account AND binIndex > :binIndex"),
+		ExpressionAttributeValues: commondynamodb.ExpressionValues{
+			":account":  &types.AttributeValueMemberS{Value: accountID},
+			":binIndex": &types.AttributeValueMemberN{Value: strconv.FormatUint(uint64(binIndex), 10)},
+		},
+		ScanIndexForward: aws.Bool(true),
+		Limit:            aws.Int32(3),
+	}
+	bins, err := s.dynamoClient.QueryWithInput(ctx, queryInput)
+	if err != nil {
+		return 0, 0, 0, fmt.Errorf("failed to query payments for account: %w", err)
+	}
+
+	usages := make([]uint64, 3)
+	for i := 0; i < len(bins) && i < 3; i++ {
+		usage, err := parseBinUsage(bins[i])
+		if err != nil {
+			return 0, 0, 0, fmt.Errorf("failed to parse bin %d usage: %w", i, err)
+		}
+		usages[i] = usage
+	}
+
+	return usages[0], usages[1], usages[2], nil
+}
+
+func (s *OffchainStore) GetLargestCumulativePayment(ctx context.Context, accountID string) (*big.Int, error) {
+	// Fetch the largest cumulative payment
+	queryInput := &dynamodb.QueryInput{
+		TableName:              aws.String(s.onDemandTableName),
+		KeyConditionExpression: aws.String("AccountID = :account"),
+		ExpressionAttributeValues: commondynamodb.ExpressionValues{
+			":account": &types.AttributeValueMemberS{Value: accountID},
+		},
+		ScanIndexForward: aws.Bool(false),
+		Limit:            aws.Int32(1),
+	}
+	payments, err := s.dynamoClient.QueryWithInput(ctx, queryInput)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query payments for account: %w", err)
+	}
+
+	if len(payments) == 0 {
+		return nil, nil
+	}
+
+	payment, err := strconv.ParseUint(payments[0]["CumulativePayments"].(*types.AttributeValueMemberN).Value, 10, 64)
```

Comment:
Yeah we are going to use `wei` onchain, but we really want to use this in terms of `gwei`, as in price per symbol will be set to ~0.5gwei. I agree `uint64` might be too small here, and we would prefer to use big.Int 🤔 ....

Reply:
maybe this fixes the issue: https://github.com/Layr-Labs/eigenda/pull/937

Reply:
Are you looking to fix this in a separate PR or this PR? This method returns a bigint but this value is read as uint64, which would be a problem if the value is greater than uint64

Reply:
This value should never be greater than uint64. Napkin math being the maximum of Uint64 is ~18.5*1e18 wei, which is 18eth. I expect that we enforce this on the contracts side; plus I'm also requesting on the contracts side to use a different unit so that we can store ~18.5*1e24 on a uint64 slot, which means it is 18*1e6 eth. 
Given this context, I don't think we need additional fixing...?